File: TP3/TP4/Ej3/Ellipse.java
package Ej3;


public class Ellipse extends Figure {
    private Point centerPoint;
    private double sMayorAxis;
    private double sMinorAxis;

    public Ellipse(Point centerPoint, double sMayotAxis, double sMinorAxis) {
        this.centerPoint = centerPoint;
        this.sMayorAxis = sMayotAxis;
        this.sMinorAxis = sMinorAxis;
    }

    @Override
    public double area() {
        return Math.PI / 4 * sMayorAxis * sMinorAxis;
    }
 
    @Override
    public double perimeter() {
        return Math.PI / 2 * (sMayorAxis + sMinorAxis);
    }
 
    @Override
    public String toString() {
        return "Elipse [Centro: %s, DMayor: %.2f, DMenor: %.2f]".formatted(centerPoint, sMayorAxis, sMinorAxis);
    }

    @Override
    public void moveNorth(double delta) {
        centerPoint.moveNorth(delta);
    }

    @Override
    public void moveSouth(double delta) {
        centerPoint.moveSouth(delta);
    }

    @Override
    public void moveWest(double delta) {
        centerPoint.moveWest(delta);
    }

    @Override
    public void moveEast(double delta) {
        centerPoint.moveEast(delta);
    }
 

    
    
    
}

File: TP8/EJ4/ellipse.rb
# frozen_string_literal: true
require_relative 'figure'

class Ellipse < Figure
  def initialize(center_point, mayor_axis, minor_axis)
    @center_point = center_point
    @mayor_axis = mayor_axis
    @minor_axis = minor_axis
  end

  def area
    10
  end

  def perimeter
    5
  end
  def to_s
    "Ellipse #{@center_point} #{@mayor_axis} #{@minor_axis}"
  end
end
